### PR TITLE
[Model][PersonaPlex] add per-row RingKV offsets and active rows

### PR DESCRIPTION
**Pull request by MrDongsls**
PLEASE FILL IN THE PR DESCRIPTION HERE.

## Purpose

W1 standalone enabler described in #7389.

This PR makes PersonaPlex Mimi streaming state row-aware so a compact batched codec can let one row skip a tick without advancing or contaminating its state:

- Make `_RingKV.end_offset` and Mimi transformer position progression per-row.
- Thread an `active: Tensor[B]` mask through Mimi encode/decode so inactive rows preserve their KV, transformer offset, and convolution carry state.
- Add mixed-active pause/resume tests that compare active rows.

## Test Plan

**vLLM Version:** `0.29.0`
**vLLM-Omni Commit:** `6c5bd318bb70075dc6f39b141f6178408b58e4e8`

- [x] CPU tests: `pytest -q --run-level=core_model -m "core_model and cpu" tests/model_executor/models/personaplex/test_ring_per_row_offsets.py
` — 4 passed
- [x] GPU mixed-offset parity: `pytest -q --run-level=core_model -m "core_model and cuda" tests/model_executor/models/personaplex/test_ring_per_row_offsets.py
` - 4 passed
- [x] Existing PersonaPlex tests: `pytest -q --run-level=core_model tests/model_executor/models/personaplex --ignore=tests/model_executor/models/personaplex/test_ring_per_row_offsets.py` — 41 passed
- [ ] Fixed-workload A/B benchmark

**BEFORE SUBMITTING:** read [CONTRIBUTING.md](https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md) and run the [precheck-pr skill](https://github.com/vllm-project/vllm-omni/blob/main/.claude/skills/precheck-pr/SKILL.md) with the code agent for a self-check against project conventions.
(anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by hsliuustc0106** [comment]
This PR touches **vllm_omni/model_executor/, tests/model_executor/** (3 files). Based on CODEOWNERS coverage of the changed files, the most-related reviewers appear to be:

> @tzhouam @gcanlin @Sy0307

Could one of you take a look when you get a chance? Thanks!

File: vllm_omni/model_executor/models/personaplex/personaplex_temporal.py [hunk 1]
```diff
@@ -90,38 +90,56 @@ def reset(self) -> None:
...
-        positions = positions.view(1, -1)  # [1, capacity]
         below = positions < self.start_offset.view(-1, 1)  # [B, capacity]
         positions = torch.where(below, torch.full_like(positions, -1), positions)
         return self.cache[0], self.cache[1], positions
 
 
+def _normalize_temporal_active(active: torch.Tensor, reference: torch.Tensor) -> torch.Tensor:
+    if active.shape != reference.shape:
+        raise ValueError(f"active must have shape {tuple(reference.shape)}, got {tuple(active.shape)}")
+    return active.to(device=reference.device, dtype=torch.bool)
```
Comment (linyueqian) [inline vllm_omni/model_executor/models/personaplex/personaplex_temporal.py:140]: [suggestion] Callers are told to pass `bool[B]`, but nothing says which device. A CPU mask lands here in a blocking `.to(device=...)` copy on every frame (same in `_normalize_active` in `personaplex_mimi.py`), which is a host sync per tick and will fail inside CUDA graph capture when the compact batched codec from #7389 gets there. Worth stating in the docstring that the mask must already live on the model device (or accept only that and raise), so the W2 batch builder keeps a persistent device-side mask instead of discovering this under capture.

**Review by linyueqian — APPROVED** [review]
Reviewed at `a89c95fb` (static read against main plus the CPU and CUDA parity tests as written; nothing executed here). The change is contained to the two PersonaPlex streaming modules and is bit-for-bit the old path when every row is active: with lockstep `end_offset` the per-row `scatter_` hits the same ring slots as the old `index_copy_`, the position mask is the same formula per row, the RoPE reshape to `[B, 1, T, 1]` collapses to the old `[1, T, 1]` broadcast, and `_fresh & active` is `_fresh` for an all-true mask, so the first-frame replicate padding and the transposed-conv bias skip are unchanged. `active=None` maps to the device-resident `_all_active`, so Stage 0 and Code2Wav keep calling `encode_frame` / `decode_frame` exactly as before. Inactive rows are inert where it matters: the gathered K/V slot is written back unchanged, `end_offset`, `prev`, `partial` and `_fresh` only move where `active` is true, and `reset_slot` now masks the row's own write head rather than the shared one. `PersonaPlexTemporalStreaming.step` has no callers inside `vllm_omni` yet, so making `active` required there is safe.

The tests are the right shape: batched rings, transformers and conv carries are compared against singletons that only step on active ticks, and inactive rows are checked against a cloned snapshot; marks match the Simple and Model Executor CUDA steps. One suggestion inline about the mask's device for the future batched callers. Base is a few days old but the merged tree is byte-identical to this head for all three files, and the general lane is green at `a89c95fb`. Approving.

File: vllm_omni/model_executor/models/personaplex/personaplex_temporal.py (hunk 1 above)
Comment (MrDongsls) [reply vllm_omni/model_executor/models/personaplex/personaplex_temporal.py:140]: Fixed in a follow-up, `active` is now a persistent device-side `bool[B]` mask and raises on device or shape mismatches.
